### DISCUSSION TITLE
Nested elements in JointJS +Dagre?

### DISCUSSION THREAD

**Discussion by sabrael** [clientIO/joint#1856] [Q&A] [answered]
I've got a small test module with JointJS + Dagre, and for the most part element positioning works exactly as I'd expect it to. The only real issue is embedding elements - I can tell that *something* is happening, because if I call cell.fitEmbeds() the parent is properly resized to fit the expected contents. However, the embedded element is treated as an unconnected node by Dagre, and is simply placed to the side of the other top-level elements in the graph. Here's the code I'm using for a minimal reproduction - simply placing a circular element inside of a square element. The image that follows is what's produced in the graph, and the second image shows the output if I don't call rect.fitEmbeds();

```
var rect = new joint.shapes.basic.Rect();
var circle = new joint.shapes.basic.Circle();

graph.addCells([rect, circle])
rect.embed(circle);
rect.fitEmbeds();

```

![embedding_issue](https://user-images.githubusercontent.com/32584035/197275363-12ef6f0b-9566-49a3-b346-a0b80ff6427e.png)

![embedding_issue2](https://user-images.githubusercontent.com/32584035/197276039-e54070ce-8ed2-4cd0-aef6-cc946403ed43.png)

I'm sure this is because I'm using Dagre for positioning, but I'm having difficulty finding any documentation / discussions re: clustering nodes with JointJS + Dagre, as opposed to Dagre + D3. If anyone could point me towards some relevant information, or better yet a working example I could play around with, it would be greatly appreciated!

**Comment by kumilingus** [comment, marked as answer]
Here's the [documentation](https://resources.jointjs.com/docs/jointjs/v3.6/joint.html#layout) for `DirectedGraph` (`dagre` wrapper).
Here's an [example with clusters](https://resources.jointjs.com/docs/jointjs/v3.6/demo/layout/DirectedGraph/clusters.html) from the docs.

**Reply by sabrael** [reply]
Thanks!
